### PR TITLE
Bug fixes and test stabilization to unblock 1.13 release

### PR DESCRIPTION
**Pull request by artursouza**
# Description

* Use "127.0.0.1" instead of "localhost" in tests - we have seen instability in resolving "localhost" to an IP address that could not respond before
* Use of DAPR_HOST_IP to force sidecar to advertise itself as "127.0.0.1" in integration-tests
* Fix IsActorLocal() to also compare port in case of localhost
* Fix reminder rebalance bug (by @elena-kolevska)

## Issue reference

Please reference the issue this PR will close: endgame

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/integration/suite/actors/healthz/endpoint/noentities.go [hunk 1]
```diff
@@ -125,10 +135,17 @@ func (n *noentities) Run(t *testing.T, ctx context.Context) {
 
 	client := util.HTTPClient(t)
 
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, fooActorURL(n.daprd), nil)
-	require.NoError(t, err)
-	resp, err := client.Do(req)
-	require.NoError(t, err)
-	assert.Equal(t, http.StatusOK, resp.StatusCode)
-	require.NoError(t, resp.Body.Close())
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
```
Comment (JoshVanL) [inline tests/integration/suite/actors/healthz/endpoint/noentities.go:138]: We shouldn't _have_ to eventually here as a request should always succeed in creating the actor after daprd reports placement ready

**Review by JoshVanL — DISMISSED** [review]

File: tests/integration/suite/actors/healthz/endpoint/noentities.go (hunk 1 above)
Comment (artursouza) [reply tests/integration/suite/actors/healthz/endpoint/noentities.go]: Reverted. It was part of the troubleshooting. Thanks.

**Comment by artursouza** [comment]
Even if all tests pass, let me re-run them a few times to make sure the flakiness is over.

File: pkg/api/grpc/proxy/handler_test.go [hunk 2]
```diff
@@ -833,7 +832,7 @@ func (s *proxyTestSuite) SetupSuite() {
 
 	clientConn, err := grpc.DialContext(
 		context.Background(),
-		strings.Replace(s.proxyListener.Addr().String(), "127.0.0.1", "localhost", 1),
+		s.proxyListener.Addr().String(), // DO NOT USE "localhost" as it does not resolve to loopback in some environments.
```
Comment (ItalyPaleAle) [inline pkg/api/grpc/proxy/handler_test.go:835]: I don't think this will work on IPv6-only networks?

**Review by elena-kolevska — APPROVED** [review]

File: pkg/api/grpc/proxy/handler_test.go (hunk 2 above)
Comment (artursouza) [reply pkg/api/grpc/proxy/handler_test.go:835]: I remember these fixing flakiness here. I will run another PR without the localhost changes for us to compare now that we have a green baseline. If ipv6 only networks is the concern (although rare for local networks), I can take this value from an env variable.

Comment (artursouza) [reply pkg/api/grpc/proxy/handler_test.go:835]: Let's compare: https://github.com/dapr/dapr/pull/7492

Comment (artursouza) [reply pkg/api/grpc/proxy/handler_test.go:835]: As I mentioned in Discord, this is only for tests and not in runtime code.

**Review by yaron2 — APPROVED** [review]

**Comment by artursouza** [comment]
I created 2 issues to follow up on this and not block the release since the issues on MacOS runner might be an environment issue but requires further investigation.

https://github.com/dapr/dapr/issues/7501
https://github.com/dapr/dapr/issues/7502

**Pull request merged**